src/streaming.ts

```diff
@@ -115,12 +122,20 @@ export class StreamProcessor {
       this.log.debug('SSE already connected, skip retry');
       return;
     }
+    // Cleanup the previous AbortController instance if it exists
+    if (this.abortController) {
+      this.abortController.abort();
+    }
+    this.abortController = new AbortController();
+    const {signal} = this.abortController;
 
     const isSecure = url.startsWith('https:');
     this.log.debug('SSE HTTP start request', url);
 
+    const appendedOptions = {...options, signal};
```

Comment:
We have to create a new object from scratch which includes the original http options + the signal to abort. Otherwise, the type checking fails, so it is merely to satisfy the typescript linter.